model-sync-gradle/src/main/kotlin/org/modelix/model/sync/gradle/tasks/GenerateAntScriptForMps.kt

```diff
@@ -0,0 +1,125 @@
+
+            <target name="clean">
+                <delete dir="${"$"}{build.mps.config.path}" />
+                <delete dir="${"$"}{build.mps.system.path}" />
+            </target>
+
+            <target name="declare-mps-tasks">
+                <taskdef resource="jetbrains/mps/build/ant/antlib.xml" classpathref="path.mps.ant.path" />
+            </target>
+
+            <target name="${if (isExport) "export" else "import"}-modules" depends="clean,declare-mps-tasks">
+                <echo message="Running ${if (isExport) "export" else "import"} of modules" />
+                <runMPS solution="ac6b4971-2a89-49fb-9c30-c2f0e85de741(org.modelix.model.sync.mps)" startClass="org.modelix.model.sync.mps.MPSRepositorySynchronizer" startMethod="${if (isExport) "export" else "import"}Repository">
+                    <library file="${mpsDependenciesPath.get()}" />
+                    <library file="${repositoryPath.get()}" />
+                    <jvmargs>
+                        <arg value="-Dmodelix.model.sync.${if (isExport) "output" else "input"}.path=${jsonDirPath.get()}" />
+                        <arg value="-Dmodelix.model.sync.${if (isExport) "output" else "input"}.modules=${includedModules.get().joinToString(",")}" />
+                        <arg value="-Dmodelix.model.sync.repo.path=${repositoryPath.get()}" />
+                        <arg value="-Didea.config.path=${"$"}{build.mps.config.path}" />
+                        <arg value="-Didea.system.path=${"$"}{build.mps.system.path}" />
+                        <arg value="-ea" />
+                        <arg value="-Xmx${mpsHeapSize.get()}" />
+                    </jvmargs>
+                </runMPS>
+            </target>
+        </project>
+            """.trimIndent(),
+        )
+    }
+
+    private fun getAntLibs(mpsHome: File): List<String> {
+        val mpsVersion = getMpsVersion(mpsHome)
+        return if (mpsVersion < "2021.2") {
+            listOf("lib/ant/lib/ant-mps.jar", "lib/log4j.jar", "lib/jdom.jar")
+        } else if (mpsVersion < "2021.3") {
+            listOf("lib/ant/lib/ant-mps.jar", "lib/util.jar")
+        } else {
+            listOf("lib/ant/lib/ant-mps.jar", "lib/util.jar", "lib/3rd-party-rt.jar")
+        }
+    }
+
+    private fun getMpsVersion(mpsHome: File): String {
+        val buildPropertiesFile = mpsHome.resolve("build.properties")
+        require(buildPropertiesFile.exists()) { "MPS build.properties file not found: ${buildPropertiesFile.absolutePath}" }
+        val buildProperties = Properties()
+        buildPropertiesFile.inputStream().use { buildProperties.load(it) }
+
+        return listOfNotNull(
+            buildProperties["mpsBootstrapCore.version.major"],
+            buildProperties["mpsBootstrapCore.version.minor"],
+            buildProperties["mpsBootstrapCore.version.eap"],
+        )
+            .map { it.toString().trim('.') }
+            .filter { it.isNotEmpty() }
+            .joinToString(".")
```

Comment:
if only minor would be `null`, this would result in a strange version being returned. Maybe this should be made safer via some assertions?